src/magicbox/MagicBox.ts

```diff
@@ -154,6 +154,8 @@ export class MagicBoxInstance {
       } else if (key === KEYBOARD.ESCAPE) {
         this.clearSuggestion();
         this.blur();
+      } else {
+        this.suggestionsManager.clearKeyboardFocusedElement();
```

Comment:
I'm curious about the difference between this path and the escape path. Here we clear the focused element, whereas in the other, we clear the suggestion. Curious what the difference is.

Also, shouldn't we add a new test for the inputManager.onkeydown to check this path?

Reply:
I think a lot of tests need to be added here, there aren't really any tests for the MagicBox. A refactor is coming for this part and I think it would be a good time at that point to add these tests.